### PR TITLE
[core] add Sekkanoki functionality to mobskill_state.cpp 

### PR DESCRIPTION
**Pull request by GodModeXI**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

- Creates functionality for Sekkanoki for mobskills. This is primarily so that trusts can utilize the ability through gambits. Videos below.
- video 1 : Sekkanoki prior to code. WS depletes to 0 regardless of status.
    - https://user-images.githubusercontent.com/99501957/197692301-efd20bb4-e387-4962-806c-71e49d364f12.mov

- video 2 : Sekkanoki after update. Uses 1000 tp and saves 2000 for self skillchain
    - https://user-images.githubusercontent.com/99501957/197692629-2031a9b1-9cc3-4668-9f29-e4bfc7e0f67a.mov

## Steps to test these changes

1. load in changes.
2. clean and rebuild server to apply core changes
3.  load up a SAM trust spell lua (AAGK is who I used) with :
     - mob:addSimpleGambit(ai.t.SELF, ai.c.TP_GTE, 2000, ai.r.JA, ai.s.SPECIFIC, xi.ja.SEKKANOKI)
     - mob:setTrustTPSkillSettings(ai.tp.CLOSER_UNTIL_TP, ai.s.HIGHEST, 3000)
5. engage in combat and use !tp 2000 to trigger the ability.
6. use !tp 3000 to trigger WS use. (and self skillchain!)

### CODE REVIEW

**Review by WinterSolstice8 — APPROVED** [review]

**Comment by WinterSolstice8** [comment]
Fixup CI and lgtm

**Review by WinterSolstice8 — CHANGES_REQUESTED** [review]
On second thought, that check should be inside that if for non-free WS. There is a corner case where sekkanoki could make free WS cost something.

**Comment by GodModeXI** [comment]
> On second thought, that check should be inside that if for non-free WS. There is a corner case where sekkanoki could make free WS cost something.

IsTPFreeSkill should be it's own if and not an else if?

**Comment by Xaver-DaRed** [comment]
```c++
void CMobSkillState::SpendCost()
{
    if (!m_PSkill->isTpFreeSkill()) // if it is NOT a TP-Free skill
    {
        if (m_PEntity->StatusEffectContainer->HasStatusEffect(EFFECT_SEKKANOKI))
        {
            m_spentTP = m_PEntity->addTP(-1000);
            m_PEntity->StatusEffectContainer->DelStatusEffect(EFFECT_SEKKANOKI);
        }
        else
        {
            m_spentTP            = m_PEntity->health.tp;
            m_PEntity->health.tp = 0;
        }
    }
}
```

File: src/map/ai/states/mobskill_state.cpp [hunk 1]
```diff
@@ -84,11 +84,22 @@ CMobSkill* CMobSkillState::GetSkill()
...
+            m_PEntity->StatusEffectContainer->DelStatusEffect(EFFECT_SEKKANOKI);
+        }
+        else
+        {
+            m_spentTP            = m_PEntity->health.tp;
+            m_PEntity->health.tp = 0;
+        }
+    }    
+
+    m_spentTP = tp;
```
Comment (Xaver-DaRed) [inline src/map/ai/states/mobskill_state.cpp]: "tp" wont exist on regular mobskills

File: src/map/ai/states/mobskill_state.cpp [hunk 2]
```diff
@@ -84,11 +84,22 @@ CMobSkill* CMobSkillState::GetSkill()
...
 {
+    auto tp = 0;
     if (!m_PSkill->isTpFreeSkill())
     {
-        m_spentTP            = m_PEntity->health.tp;
-        m_PEntity->health.tp = 0;
-    }
+        if (m_PEntity->StatusEffectContainer->HasStatusEffect(EFFECT_SEKKANOKI))
+        {
+            tp = m_PEntity->addTP(-1000);
```
Comment (Xaver-DaRed) [inline src/map/ai/states/mobskill_state.cpp]: why define a new variable (tp) when m_spentTP already exists?

Comment (GodModeXI) [reply src/map/ai/states/mobskill_state.cpp]: ahh I see what you mean now.

**Review by Xaver-DaRed — APPROVED** [review]

**Review by WinterSolstice8 — APPROVED** [review]

**Review by claywar — APPROVED** [review]

**Pull request merged**